mod-audit-server/src/main/java/org/folio/util/PubSubClientUtils.java

```diff
@@ -0,0 +1,238 @@
+          String msg = String.format("Module %s was not unregistered as '%s' in PubSub. HTTP status: %s", moduleId, moduleRole, response.result().statusCode());
+          LOGGER.error(msg);
+          future.completeExceptionally(new ModuleUnregistrationException(msg));
+        }
+
+      });
+    } catch (Exception var6) {
+      LOGGER.error("Module was not unregistered as '{}' in PubSub.", moduleRole, var6);
+      promise.fail(var6);
+      future.completeExceptionally(var6);
+    }
+
+    return future;
+  }
+
+  static DescriptorHolder readMessagingDescriptor() throws Throwable {
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    try {
+      MessagingDescriptor messagingDescriptor = objectMapper.readValue(getMessagingDescriptorInputStream(), MessagingDescriptor.class);
+      return (new DescriptorHolder()).withPublisherDescriptor((new PublisherDescriptor()).withModuleId(constructModuleName()).withEventDescriptors(messagingDescriptor.getPublications())).withSubscriberDescriptor((new SubscriberDescriptor()).withModuleId(constructModuleName()).withSubscriptionDefinitions(messagingDescriptor.getSubscriptions()));
+    } catch (JsonMappingException | JsonParseException var3) {
+      String errorMessage = "Can not read messaging descriptor, cause: " + var3.getMessage();
+      LOGGER.error(errorMessage);
+      throw new IllegalArgumentException(var3);
+    }
+  }
+
+  private static InputStream getMessagingDescriptorInputStream() throws Throwable {
+    return (InputStream)((Optional) Optional.ofNullable(System.getProperty("messaging_config_path")).flatMap(PubSubClientUtils::getFileInputStreamByParentPath).or(() -> getFileInputStreamFromClassPath("MessagingDescriptor.json"))).orElseThrow(() -> new MessagingDescriptorNotFoundException("Messaging descriptor file 'MessagingDescriptor.json' not found"));
+  }
+
+  private static Optional<InputStream> getFileInputStreamByParentPath(String parentPath) {
+    if (Paths.get(parentPath).isAbsolute()) {
+      return getFileInputStreamByAbsoluteParentPath(parentPath);
+    } else {
+      String fullRelativeFilePath = parentPath + File.separatorChar + "MessagingDescriptor.json";
+      return getFileInputStreamFromClassPath(fullRelativeFilePath);
+    }
+  }
+
+  private static Optional<InputStream> getFileInputStreamByAbsoluteParentPath(String absoluteParentPath) {
+    File file = new File(absoluteParentPath, "MessagingDescriptor.json");
+
+    try {
+      return Optional.of(new FileInputStream(file));
+    } catch (FileNotFoundException var3) {
+      return Optional.empty();
+    }
+  }
+
+  private static Optional<InputStream> getFileInputStreamFromClassPath(String path) {
+    String preparedPath = path.replace('\\', '/');
+    InputStream fileStream = org.folio.util.pubsub.PubSubClientUtils.class.getClassLoader().getResourceAsStream(preparedPath);
+    return fileStream == null ? Optional.empty() : Optional.of(fileStream);
+  }
+
+  public static String constructModuleName() {
```

Comment:
The problem with [issue](https://issues.folio.org/browse/FOLIO-3183) was due to this method: instead of ModuleName and RmbVersion PomReader was used (PomReader is no longer supported in the new 33.0.0 version).